### PR TITLE
Adding Signature API Samples

### PR DESCRIPTION
**Pull request by emtencza**
Adding samples in ScriptLab for the Signature APIs including:

1. isClientSignatureEnabledAsync
2. disableClientSignatureAsync
3. getComposeTypeAsync
4. setSignatureAsync

### CODE REVIEW

**Comment by emtencza** [comment]
Waiting to merge these changes when Office.js changes for the Signature APIs are available on the beta CDN.

File: view/outlook.json [hunk 1]
```diff
@@ -45,5 +45,6 @@
   "outlook-tokens-and-service-calls-make-ews-request-async": "https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/85-tokens-and-service-calls/make-ews-request-async.yaml",
   "outlook-tokens-and-service-calls-send-message-using-make-ews-request-async": "https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/85-tokens-and-service-calls/send-message-using-make-ews-request-async.yaml",
   "outlook-tokens-and-service-calls-item-id-compose": "https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/85-tokens-and-service-calls/item-id-compose.yaml",
-  "outlook-other-item-apis-get-set-item-subject": "https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/90-other-item-apis/get-set-item-subject.yaml"
+  "outlook-other-item-apis-get-set-item-subject": "https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/90-other-item-apis/get-set-item-subject.yaml",
+  "outlook-signature-work-with-client-signatures": "https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/preview-apis-signature/work-with-client-signatures.yaml"
```
Comment (ElizabethSamuel-MSFT) [inline view/outlook.json]: The preview APIs should go into a folder called `99-preview-apis`

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    Mailbox: '1.1'
+script:
+    content: |
+        $("#isClientSignatureEnabled").click(isClientSignatureEnabled);
+        $("#disableClientSignature").click(disableClientSignature);
+        $("#getComposeType").click(getComposeType);
+        $("#setSignature").click(setSignature);
+
+        function isClientSignatureEnabled() {
+          // Check if the client signature is currently enabled
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: We usually end code comments with a period e.g.
```suggestion
          // Check if the client signature is currently enabled.
```

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
+id: outlook-signature-work-with-client-signatures
+name: Work with Client Signatures
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: I think these APIs only work apply to compose mode, right? If so, then name should be something like:
```suggestion
name: Work with Client Signatures (Compose)
```

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+description: 'Check if the client signature is enabled, disable the client signature, get the compose type, and set a signature.'
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]:
```suggestion
description: 'Check if the client signature is enabled, disable the client signature, get the compose type, and set a signature in Compose mode.'
```

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            if (asyncResult.status === Office.AsyncResultStatus.Succeeded) {
+              console.log("disableClientSignatureAsync succeeded");
+            } else {
+              console.error(asyncResult.error);
+            }
+          });
+        }
+
+        function getComposeType() {
+          // Get the compose type of the current item (compose only)
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: If the entire set of APIs only apply in compose mode, then you don't need to specify here.
```suggestion
          // Get the compose type of the current item.
```

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+              console.error(asyncResult.error);
+            }
+          });
+        }
+    language: typescript
+template:
+    content: |-
+        <section class="ms-font-m">
+            <p>This sample shows how to check if the client signature is enabled, disable the client signature, get the compose type, and set the signature.</p>
+        </section>
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: Insert a statement here to indicate required mode e.g. `<p><b>Required mode</b>: Compose</p>`

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+        <section class="samples ms-font-m">
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: Ensure that the html is indented appropriately

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+        section.samples .ms-Button, section.setup .ms-Button {
+            display: block;
+            margin-bottom: 5px;
+            margin-left: 20px;
+            min-width: 80px;
+        }
+    language: css
+libraries: |
+    https://appsforoffice.microsoft.com/lib/beta/hosted/office.js
```
Comment (ElizabethSamuel-MSFT) [inline samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: When the APIs are available in beta, let me know so I can add details to the office-js-preview d.ts in DefinitelyTyped

Comment (emtencza) [reply samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: It is not on the beta CDN yet. I will let you know when it is deployed.

File: playlists/outlook.yaml [hunk 9]
```diff
@@ -439,3 +439,14 @@
     https://raw.githubusercontent.com/<ACCOUNT>/<REPO>/<BRANCH>/samples/outlook/90-other-item-apis/get-set-item-subject.yaml
   group: Other Item APIs
   api_set: {}
+- id: outlook-work-with-client-signatures
+  name: Work With Client Signatures (Compose)
+  fileName: work-with-client-signatures.yaml
+  description: >-
+    Check if the client signature is enabled, disable the client signature, get
+    the compose type, and set a signature in Compose mode.
```
Comment (danielebercovici) [inline playlists/outlook.yaml]: Maybe add that is for Messages and not Calendar?

File: samples/outlook/preview-apis-signature/work-with-client-signatures.yaml (hunk 8 above)
Comment (exextoc) [reply samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: These APIs are now available in beta.

Comment (emtencza) [reply samples/outlook/preview-apis-signature/work-with-client-signatures.yaml]: These APIs are now available in beta

**Comment by ElizabethSamuel-MSFT** [comment]
Per our offline discussion, I'll do the mapping update for the reference docs in a subsequent PR.

**Pull request merged**